src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Formatting/FormattingContext.cs

```diff
@@ -191,13 +191,23 @@ public async Task<FormattingContext> WithTextAsync(SourceText changedText)
                 }
                 else
                 {
-                    // Couldn't find a corresponding FormattingSpan.
+                    // Couldn't find a corresponding FormattingSpan. Happens if it is a 0 length line.
+                    // Let's create a 0 length span to represent this and default it to HTML.
+                    var placeholderSpan = new FormattingSpan(
+                        new Language.Syntax.TextSpan(total + nonWsChar, 0),
+                        new Language.Syntax.TextSpan(total + nonWsChar, 0),
+                        FormattingSpanKind.Markup,
+                        FormattingBlockKind.Markup,
+                        indentationLevel: 0,
+                        isInClassBody: false);
+
                     result.Indentations[i] = new IndentationContext
                     {
                         Line = i,
-                        IndentationLevel = -1,
```

Comment:
I would have expected that you would need to remove the "special case" understanding of "-1" no?

Reply:
After this change, there is no other way to have a -1 indentation anywhere. Turns out I was not using this anywhere so there is no special case to remove. Hope I understood your question correctly.